### PR TITLE
Add templates to Github repository

### PR DESCRIPTION
**Pull request by Springo**
## Goal
Make Github repository more organized.

## Things To Do
- [x] Add pull request template.
- [x] Add issue template.
- [x] Add contributing readme.
- [x] Finish editing templates.

### CODE REVIEW

**Comment by Springo** [comment]
@irenfro Could you review the issue template I added? Afterwards, I'll leave the CONTRIBUTING.md file to you if you've got ideas for that before we merge this PR.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Description
```
Comment (irenfro) [inline .github/ISSUE_TEMPLATE.md]: We should split the description up into 2 fields: Current Behavior and Expected Behavior

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+*Describe problem and desired outcome.*
+
+## Solution Ideas
```
Comment (irenfro) [inline .github/ISSUE_TEMPLATE.md]: Before this we need something like: Minimal reproduction of the problem with instructions

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (irenfro) [inline .github/ISSUE_TEMPLATE.md]: We should split the description up into 2 fields: Current Behavior and Expected Behavior

**Review by irenfro — CHANGES_REQUESTED** [review]
@Springo just look at these two and I think you should be good

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# Contributing to Project-Euler-Library-Code
+ - [Pull Request Guidelines](#pull-request-guidelines)
+ - [Rebasing / Squashing Commits](#rebasing-squashing-commits)
+ - [Commit Message Format](#commit-message-format)
+ 
+## Pull Request Guidelines
```
Comment (Springo) [inline CONTRIBUTING.md:6]: I think the pull request section should be based off our template (which we can change).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+After one or more core team members have given it a **Thumbs Up**:
+1. [Rebase](#rebasing-squashing-commits) the branch locally into a single commit following the [commit message format](#commit-message-format).
+2. Force Push the local, rebased branch back to Git.
+3. If the Merge Request still displays the old commits, then close and re-open the request.
+ 
+## Rebasing / Squashing Commits
+1. Fetch the latest changes by going to VCS->Git->Fetch
+2. Check out the latest version of master to your local branches.
+3. Go to VCS->Git->Rebase.
```
Comment (Springo) [inline CONTRIBUTING.md]: I would prefer that our method of merging in pull requests doesn't require another software. I think all of these instructions should be written with respect to Git Bash.

Comment (Springo) [reply CONTRIBUTING.md]: If that's too complicated, I think we might want to use a simpler method of merging in pull requests for a standard pull request.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Sample Commit
+```
+fix: sytax error on function
```
Comment (Springo) [inline CONTRIBUTING.md]: Whoops, spelling error here.

**Comment by Springo** [comment]
@irenfro Your changes look good to me, do you approve of my earlier changes to the issue and PR templates?

**Comment by irenfro** [comment]
@Springo yep they look good to me

**Pull request merged**